libs/partners/huggingface/langchain_huggingface/llms/huggingface_endpoint.py

```diff
@@ -145,19 +146,18 @@ def build_extra(cls, values: Dict[str, Any]) -> Dict[str, Any]:
             )
 
         values["model_kwargs"] = extra
-
-        values["endpoint_url"] = get_from_dict_or_env(
-            values, "endpoint_url", "HF_INFERENCE_ENDPOINT", None
-        )
-
-        if values["endpoint_url"] is None and "repo_id" not in values:
+        if "endpoint_url" not in values and "repo_id" not in values:
             raise ValueError(
                 "Please specify an `endpoint_url` or `repo_id` for the model."
             )
-        if values["endpoint_url"] is not None and "repo_id" in values:
+        if "endpoint_url" in values and "repo_id" in values:
             raise ValueError(
                 "Please specify either an `endpoint_url` OR a `repo_id`, not both."
             )
+        if "repo_id" not in values:
+            values["endpoint_url"] = get_from_dict_or_env(
```

Comment:
We lost validation on repo_id being unspecified and endpoint_url being unspecified

Reply:
Nope, the validation is happening above. See Line 149. @eyurtsev